spec/matchers/discussion_matchers.rb

```diff
@@ -0,0 +1,96 @@
+require "rspec/expectations"
+
+module DiscussionMatchers
+  extend RSpec::Matchers::DSL
+
+  matcher :have_correct_discussion_title_on_page do |discussion_title|
+    match do |actual|
+      expect(actual).to have_text(discussion_title)
+    end
+    failure_message do |actual|
+      "expected that the discussion title '#{discussion_title}' would be present on the page."
+    end
+  end
+
+  matcher :have_correct_discussion_body_on_page do |discussion_body|
+    match do |actual|
+      expect(actual).to have_text(discussion_body)
+    end
+    failure_message do |actual|
+      "expected that the discussion body '#{discussion_body}' would be present on the page."
+    end
+  end
+
+  matcher :have_correct_discussion_message_for_action do |action|
+    expected_text = "Discussion was successfully #{action}."
+    match do |actual|
+      expect(actual).to have_text(expected_text)
+    end
+    failure_message do |actual|
+      "expected that the message '#{expected_text}' would be present on the page for the appropriate action."
+    end
+  end
+
+  matcher :have_expected_discussion_buttons do
+    match do
+      find_button "Ban"
+      find_button "Pin"
+      find_button "Lock"
+      find_button "Edit"
+      find_button "Delete"
+    end
+    failure_message do
+      "expected that the buttons `Edit` and `Create Discussion` exist on the page."
```

Comment:
The failure message does not align with the matcher logic.  This also doesn't correlate directly with discussion buttons themselves other than via the name of the matcher, and could trigger on Comment buttons.  This seems more like an array of buttons that should be in the discussion spec file itself that calls `find_button` on each, rather than attempt to build a semantic matcher.